test/Sql/SelectTest.php

```diff
@@ -1397,4 +1399,21 @@ public function providerData()
             [$select52, $sqlPrep52, [],    $sqlStr52, $internalTests52],
         ];
     }
+
+    public function testPropagateAdapterWithGetSqlStringThatAlreadyBroughtBySqlObject()
+    {
+        if (extension_loaded('mysqli')) {
```

Comment:
Specifically, you should not be using a specific extension and thus providing a mock for the adapter rather than utilizing it directly as per this example.